flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/DataSetMetaInfo.java

```diff
@@ -44,6 +48,14 @@ public int getNumTotalPartitions() {
         return numTotalPartitions;
     }
 
+    public void addShuffleDescriptors(Set<ShuffleDescriptor> shuffleDescriptors) {
+        this.shuffleDescriptors.addAll(shuffleDescriptors);
```

Comment:
We use set here to track shuffle descriptors. Does that mean we force every plugin implementation to must Implement a hashCode and equals method. If so, I think it makes it easy to cause mistake.